stubs/urllib3/urllib3/connectionpool.pyi

```diff
@@ -29,48 +34,52 @@ Retry = retry.Retry
 Timeout = timeout.Timeout
 get_host = url.get_host
 
+_Timeout: TypeAlias = Timeout | float
+_Retries: TypeAlias = Retry | bool | int
+
 xrange: Any
-log: Any
+log: Logger
 
 class ConnectionPool:
-    scheme: Any
-    QueueCls: Any
-    host: Any
-    port: Any
-    def __init__(self, host, port=...) -> None: ...
-    def __enter__(self): ...
-    def __exit__(self, exc_type, exc_val, exc_tb): ...
-    def close(self): ...
+    scheme: ClassVar[str | None]
+    QueueCls: ClassVar[Any]
+    host: str
+    port: int | None
+    def __init__(self, host: str, port: int | None = ...) -> None: ...
+    def __enter__(self: Self) -> Self: ...
+    def __exit__(self, exc_type, exc_val, exc_tb) -> bool: ...
```

Comment:
The return type here is quite important -- both mypy and pyright do quite heavy special-casing of `__exit__` methods. If they see an `__exit__` method that's annotated as returning `bool`, they'll assume it's an `__exit__` method that can sometimes swallow exceptions. But this method _can't_, because it always returns `False`, never `True`.

```suggestion
    def __exit__(self, exc_type: type[BaseException] | None, exc_val: BaseException | None, exc_tb: TracebackType | None) -> Literal[False]: ...
```

Reply:
Ah, I had seen https://github.com/python/typeshed/issues/7258 and thought from the title that this was related to literal's in this sense (my bad on not reading the issue, just the title). Sorry about that. I'll fix that in a moment